openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -0,0 +1,452 @@
+    ref = ContentLibrary.objects.get(slug=library_key.slug)
+    # Make sure the proposed ID will be valid:
+    validate_unicode_slug(definition_id)
+    # Ensure the XBlock type is valid and installed:
+    XBlock.load_class(block_type)  # Will raise an exception if invalid
+    # Make sure the new ID is not taken already:
+    new_usage_id = definition_id  # Since this is a top level XBlock, usage_id == definition_id
+    usage_key = LibraryUsageLocatorV2(library_slug=library_key.slug, block_type=block_type, usage_id=new_usage_id)
+    library_context = get_learning_context_impl(usage_key)
+    if library_context.definition_for_usage(usage_key) is not None:
+        raise LibraryBlockAlreadyExists("An XBlock with ID '{}' already exists".format(new_usage_id))
+
+    new_definition_xml = '<{}/>'.format(block_type)  # xss-lint: disable=python-wrap-html
+    path = "{}/{}/definition.xml".format(block_type, definition_id)
+    # Write the new XML/OLX file into the library bundle's draft
+    draft = get_or_create_bundle_draft(ref.bundle_uuid, DRAFT_NAME)
+    write_draft_file(draft.uuid, path, new_definition_xml)
+    # Clear the bundle cache so everyone sees the new block immediately:
+    BundleCache(ref.bundle_uuid, draft_name=DRAFT_NAME).clear()
+    # Now return the metadata about the new block:
+    return get_library_block(usage_key)
+
+
+def delete_library_block(usage_key):
+    """
+    Delete the specified block from this library (and any children it has).
+
+    If this is the only usage of the block's definition within this library,
+    both the definition and the usage will be deleted. If this is only one of
+    several usages, the definition will be kept. Usages by linked bundles are
+    ignored and will not prevent deletion of the definition.
+
+    If the usage points to a definition in a linked bundle, the usage will be
+    deleted but the link and the linked bundle will be unaffected.
+    """
+    assert isinstance(usage_key, LibraryUsageLocatorV2)
+    library_context = get_learning_context_impl(usage_key)
+    library_ref = ContentLibrary.objects.get(slug=usage_key.library_slug)
+    def_key = library_context.definition_for_usage(usage_key)
+    if def_key is None:
+        raise ContentLibraryBlockNotFound(usage_key)
+    lib_bundle = LibraryBundle(library_ref.slug, library_ref.bundle_uuid, draft_name=DRAFT_NAME)
+    # Create a draft:
+    draft_uuid = get_or_create_bundle_draft(def_key.bundle_uuid, DRAFT_NAME).uuid
+    # TODO: If the definition lives in a linked bundle, just delete the usage
+    # This usage's definition has only the one usage.
+    if usage_key not in lib_bundle.get_top_level_usages():
+        # This is not a top-level block.
+        # First need to modify the parent to remove this block as a child.
+        raise NotImplementedError
+        # Then proceed to delete the definition...
+    # Next, recursively call delete_library_block(...) on each child usage
+    # TODO: delete children
```

Comment:
Same question here, is this just not written yet or is this a change we should make in the future?